### PR TITLE
fix: Try all partition_writes even if one in the middle fails

### PR DESCRIPTION
**Pull request by carols10cents**
`store_sequenced_entry` loops over each `partitioned_write` part of a `sequenced_entry` like this:

```
        if let Some(partitioned_writes) = sequenced_entry.partition_writes() {
            for write in partitioned_writes {
                ...
```

then for each table in each write, it gets or creates the partition from the catalog that the write should go in, gets a write lock to the partition, gets or creates an open chunk in the partition, gets the mutable buffer for that chunk, and writes a table batch-- which can fail and return out of the whole loop. 

So if adding a table batch into the mutable buffer fails, whatever writes *before* that batch in the `SequencedEntry` will have been written, and the ones *after* that batch won't be attempted.

In other words:

If we have a batch of writes and there is an error writing one, the result is:

```
write1 (OK)
write2 (OK)
write3 (ERR)
write4 (Not even tried)
```

And we would expect the result to be:

```
write1 (OK)
write2 (OK)
write3 (ERR)
write4 (OK -- assuming it has no errors)
```

This adds a test and a fix so that we get the expected behavior.

### CODE REVIEW

**Comment by carols10cents** [comment]
```
error: this operation will always return zero. This is likely not the intended outcome
    --> server/src/db.rs:1388:13
     |
1388 |             nanoseconds_per_hour * 0 + 1,
     |             ^^^^^^^^^^^^^^^^^^^^^^^^
```

I don't have to explain my art to you, clippy

File: server/src/db.rs [hunk 1]
```diff
@@ -1339,6 +1358,75 @@ mod tests {
...
+            SequencedEntry::new_from_sequence(Sequence { id: 1, number: 2 }, entry).unwrap(),
+        );
+
+        // This should succeed and start chunks in the MUB
+        db.store_sequenced_entry(sequenced_entry).unwrap();
+
+        // 3 more lines that should go in the 3 partitions/chunks.
+        // Line 1 has the same schema and should end up in the MUB.
+        // Line 2 has a different schema than line 1 and should error
+        // Line 3 has the same schema as line 1 and should end up in the MUB.
```
Comment (alamb) [inline server/src/db.rs:1389]: ❤️  for the comments  -- it makes the PR really easy to follow

File: server/src/db.rs [hunk 2]
```diff
@@ -1339,6 +1358,75 @@ mod tests {
...
+        let lp = format!(
+            "foo,t1=delta iv=1i {}
+             foo t1=10i {}
+             foo,t1=important iv=1i {}",
```
Comment (alamb) [inline server/src/db.rs:1393]: 😆

**Review by alamb — APPROVED** [review]
Looks great to me. The test is Top Notch 👌 

Note this might conflict with @pauldix 's changes in https://github.com/influxdata/influxdb_iox/pull/1801

File: server/src/db.rs [hunk 3]
```diff
@@ -976,7 +985,11 @@ impl Db {
                                 .context(WriteEntry {
                                     partition_key,
                                     chunk_id,
-                                })?;
+                                })
+                            {
+                                errors.push(e);
```
Comment (crepererum) [inline server/src/db.rs:990]: I'm wondering if this is a DoS vector, aka if the error structs are way larger than the actual entries and therefore might OOM the node if we store too many. So we might also just collect the first X errors and a flag if there where more than X errors (with an X=10 or so).

Comment (carols10cents) [reply server/src/db.rs:990]: Ooh good call -- I can add a limit here.

Comment (alamb) [reply server/src/db.rs:990]: I am not sure how important it is to try and protect against such an attack

the size of the erorr struct is likely to be pretty small compared the the incoming data i would think, and there is some limit on the incoming request.

There are a lot of other ways to DoS IOx than causing a bunch of errors

Comment (carols10cents) [reply server/src/db.rs:990]: I think it's more likely that we'd see an *unintentional* DoS rather than an attack-- let's say we try to write in a whole bunch of historical data in a whole bunch of partitions, and we messed up something somewhere and all the writes in all the partitions in all the world have a schema mismatch, and we can't even debug that problem because collecting all these errors causes an OOM. 

It's a pretty small change, I'll push it up and yinz can decide if you want it or not on Monday ❤️

**Review by crepererum — APPROVED** [review]

File: server/src/db.rs (hunk 3 above)
Comment (alamb) [reply server/src/db.rs]: I guess my point was that the size of the errors compared to the size of the data is likely to be quite tiny. I don't have a strong opinion. Thanks for getting this in
